### PR TITLE
Fixes #2098 too many results with wildcards

### LINKED ISSUE #2098: Too many results with wildcards
E.g the search for `getAllMoleculesMatching("Organism|Liver|Periportal|**|CYP3A4*", sim)` returns also 
**Alfentanil-CYP3A4-1st order CL Metabolite** - which is wrong

When I look for `getAllMoleculesMatching("Organism|Liver|Periportal|**|CYP3A4", sim)` 
(removed the wildcard at the end of the search path) - the result is as expected

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

**Comment by msevestre** [comment]
@Yuri05 So the problem was that the path `Organism|Liver|Periportal|**|CYP3A4*` was with a wildcard BEFORE the CYP". The implementation was wrong in this case as it was trying to match anything before, and the | was optional. That being said, I had to change some of the behavior (See comments)

File: tests/OSPSuite.R.Tests/Services/ContainerTaskSpecs.cs [hunk 1]
```diff
@@ -165,8 +169,8 @@ public class When_resolving_all_parameters_recursively_from_a_container : concer
       [Observation]
       public void should_return_the_matching_parameters()
       {
-         sut.AllParametersMatching(_organism, Constants.WILD_CARD_RECURSIVE).ShouldOnlyContain(_organism.GetAllChildren<IParameter>());
```
Comment (msevestre) [inline tests/OSPSuite.R.Tests/Services/ContainerTaskSpecs.cs:168]: This is not ok anymore
** is meant to replace tree structure. so if I Want everything it should be **|* e.g. all possible level and all names). 
I think this was a mistake

File: tests/OSPSuite.R.Tests/Services/ContainerTaskSpecs.cs [hunk 2]
```diff
@@ -129,7 +132,8 @@ public class When_resolving_all_parameters_of_a_container_with_a_container_name_
       public void should_return_the_matching_parameters()
       {
          sut.AllParametersMatching(_organism, pathFrom($"Liv{Constants.WILD_CARD}", INTRACELLULAR, $"Vol{Constants.WILD_CARD}")).ShouldOnlyContain(_volumeLiverCell);
-         sut.AllParametersMatching(_organism, pathFrom($"{Constants.WILD_CARD}Liv", INTRACELLULAR, Constants.Parameters.VOLUME)).ShouldBeEmpty();
+         sut.AllParametersMatching(_organism, pathFrom($"{Constants.WILD_CARD}Liv", INTRACELLULAR, VOLUME)).ShouldBeEmpty();
+         sut.AllParametersMatching(_organism, pathFrom(_liver.Name, Constants.WILD_CARD_RECURSIVE, $"{VOLUME}{Constants.WILD_CARD}")).ShouldOnlyContain(_volumeLiver, _volumeLiverCell);
```
Comment (msevestre) [inline tests/OSPSuite.R.Tests/Services/ContainerTaskSpecs.cs:136]: this is your use case here "**|Vol*"

File: tests/OSPSuite.R.Tests/Services/ContainerTaskSpecs.cs [hunk 3]
```diff
@@ -165,8 +169,8 @@ public class When_resolving_all_parameters_recursively_from_a_container : concer
       [Observation]
       public void should_return_the_matching_parameters()
       {
-         sut.AllParametersMatching(_organism, Constants.WILD_CARD_RECURSIVE).ShouldOnlyContain(_organism.GetAllChildren<IParameter>());
-         sut.AllParametersMatching(_liver, Constants.WILD_CARD_RECURSIVE).ShouldOnlyContain(_liver.GetAllChildren<IParameter>());
+         sut.AllParametersMatching(_organism, pathFrom(Constants.WILD_CARD_RECURSIVE, Constants.WILD_CARD)).ShouldOnlyContain(_organism.GetAllChildren<IParameter>());
+         sut.AllParametersMatching(_liver, pathFrom(Constants.WILD_CARD_RECURSIVE, Constants.WILD_CARD)).ShouldOnlyContain(_liver.GetAllChildren<IParameter>());
```
Comment (Yuri05) [inline tests/OSPSuite.R.Tests/Services/ContainerTaskSpecs.cs:173]: yep, that's exactly as expected :)

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**